src/Microsoft.SqlTools.ServiceLayer/EditData/Session.cs

```diff
@@ -110,6 +114,29 @@ public long CreateRow()
         }
 
         /// <summary>
+        /// Commits the edits in the cache to the database and then to the associated result set of
+        /// this edit session. This is launched asynchronously.
+        /// </summary>
+        /// <param name="connection">The connection to use for executing the query</param>
+        /// <param name="successHandler">Callback to perform when the commit process has finished</param>
+        /// <param name="errorHandler">Callback to perform if the commit process has failed at some point</param>
+        public void CommitEdits(DbConnection connection, Func<Task> successHandler, Func<Exception, Task> errorHandler)
```

Comment:
I don't find the separation of CommitEdits and CommitEditsInternal particularly useful because they are both pretty short. Unless there is a test case that has a good reason to bypass the checks, I personally would merge the. More of a personal preference, so it's up to you.